src/gather.jl

```diff
@@ -22,20 +25,21 @@ or multiple `dst` columns.
 
 See [`gather`](@ref) for an allocating version.
 """
+gather!(dst::AbstractArray, src::AbstractArray, idx::AbstractArray{<:IntOrIntTuple}) = gather!(dst, src, CartesianIndex.(idx))
```

Comment:
Indexing with an array of integers is by far the most common use case (in fact it is the only one implemented in ScatterNNlib). Are we sure we want to take the performance impact of the unnecessary allocation `CartesianIndex.(idx)` just to support CartesianIndexes and avoid code duplication?

Reply:
I will rethink about it.